packages/codemods/src/transforms/styled-components-to-compiled/styled-components-to-compiled.tsx

```diff
@@ -53,40 +54,55 @@ const transformer = (fileInfo: FileInfo, api: API, options: Options): string =>
     allowedImportSpecifierNames: imports.styledComponentsSupportedImportNames,
   });
 
-  const taggedTemplateExpressionsWithAttrs = collection.find(
-    j.TaggedTemplateExpression,
-    ({ tag }: TaggedTemplateExpression) => {
-      if (tag.type === 'CallExpression') {
-        const { callee } = tag;
-
-        return (
-          callee.type === 'MemberExpression' &&
-          callee.object.type === 'MemberExpression' &&
-          callee.object.object.type === 'Identifier' &&
-          callee.object.object.name === 'styled' &&
-          callee.property.type === 'Identifier' &&
-          callee.property.name === 'attrs'
-        );
-      }
-
-      return false;
-    }
-  );
-
-  if (taggedTemplateExpressionsWithAttrs.length) {
-    convertStyledAttrsToComponent({
-      j,
-      plugins,
-      expressions: taggedTemplateExpressionsWithAttrs,
-    });
-  }
-
   applyVisitor({
     plugins,
     originalProgram,
     currentProgram: collection.find(j.Program).get(),
   });
 
+  const styledImport = collection.find(
+    j.ImportSpecifier,
+    (specifier) => specifier.imported.name === 'styled'
+  );
+
+  styledImport.forEach((styledImport) => {
+    const isCompiledImport =
+      styledImport.parentPath.parentPath.value.source.value === COMPILED_IMPORT_PATH;
+
+    if (isCompiledImport) {
+      const compiledLocalStyledName = styledImport.value.local?.name || 'styled';
+
+      const taggedTemplateExpressionsWithAttrs = collection.find(
+        j.TaggedTemplateExpression,
+        ({ tag }: TaggedTemplateExpression) => {
+          if (tag.type === 'CallExpression') {
+            const { callee } = tag;
+
+            return (
+              callee.type === 'MemberExpression' &&
+              callee.object.type === 'MemberExpression' &&
+              callee.object.object.type === 'Identifier' &&
```

Comment:
same as above

Reply:
I think this one should be fine because above we check that `callee.object` is a member, which means `callee.object.object` will exist.

Reply:
we do all the previous checks to unsure the object has all necessary fields. so, I'd say it' pretty safe. any ideas how we can make it better?

Reply:
yeah looking at this one closer, the types look safe 👍